internal/cli/docker_prune.go

```diff
@@ -0,0 +1,101 @@
+package cli
+
+import (
+	"context"
+	"time"
+
+	"github.com/opentracing/opentracing-go"
+	"github.com/spf13/cobra"
+
+	"github.com/windmilleng/tilt/internal/docker"
+	"github.com/windmilleng/tilt/internal/minikube"
+
+	"github.com/windmilleng/tilt/internal/k8s"
+	"github.com/windmilleng/tilt/pkg/logger"
+)
+
+type dockerPruneCmd struct {
+	untilStr string
+}
+
+func (c *dockerPruneCmd) register() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "docker-prune",
+		Short: "run docker prune as tilt does",
+	}
+
+	cmd.Flags().StringVar(&c.untilStr, "until", "12h", "max age of image to keep (as go duration string, e.g. 1h30m, 12h")
+
+	return cmd
+}
+
+func (c *dockerPruneCmd) run(ctx context.Context, args []string) error {
+	// a := analytics.Get(ctx)
+	// a.Incr("cmd.dockerPrune", map[string]string{
+	// 	"watch": fmt.Sprintf("%v", c.watch),
+	// 	"mode":  string(updateModeFlag),
+	// })
+	// a.IncrIfUnopted("analytics.up.optdefault")
+	// defer a.Flush(time.Second)
+
+	span, ctx := opentracing.StartSpanFromContext(ctx, "Up")
+	defer span.Finish()
+
+	// tags := tracer.TagStrToMap(c.traceTags)
+
+	// for k, v := range tags {
+	// 	span.SetTag(k, v)
+	// }
+
+	deferred := logger.NewDeferredLogger(ctx)
+	ctx = redirectLogs(ctx, deferred)
+
+	dCli, err := provideDockerClient(ctx)
```

Comment:
i wonder if we should print out which docker instance this is talking to, since without executing the Tiltfile, you can't be 100% certain that you're talking to the right one (though it will mostly be ok, `tilt doctor` has a similar problem, which is why it prints out the command)